official/mnist/mnist_eager.py

```diff
@@ -159,50 +161,38 @@ def main(argv):
       checkpoint.save(checkpoint_prefix)
 
 
-class MNISTEagerArgParser(argparse.ArgumentParser):
-  """Argument parser for running MNIST model with eager training loop."""
-
-  def __init__(self):
-    super(MNISTEagerArgParser, self).__init__(parents=[
-        parsers.EagerParser(),
-        parsers.ImageModelParser()])
-
-    self.add_argument(
-        '--log_interval', '-li',
-        type=int,
-        default=10,
-        metavar='N',
-        help='[default: %(default)s] batches between logging training status')
-    self.add_argument(
-        '--output_dir', '-od',
-        type=str,
-        default=None,
-        metavar='<OD>',
-        help='[default: %(default)s] Directory to write TensorBoard summaries')
-    self.add_argument(
-        '--lr', '-lr',
-        type=float,
-        default=0.01,
-        metavar='<LR>',
-        help='[default: %(default)s] learning rate')
-    self.add_argument(
-        '--momentum', '-m',
-        type=float,
-        default=0.5,
-        metavar='<M>',
-        help='[default: %(default)s] SGD momentum')
-    self.add_argument(
-        '--no_gpu', '-nogpu',
-        action='store_true',
-        default=False,
-        help='disables GPU usage even if a GPU is available')
-
-    self.set_defaults(
-        data_dir='/tmp/tensorflow/mnist/input_data',
-        model_dir='/tmp/tensorflow/mnist/checkpoints/',
-        batch_size=100,
-        train_epochs=10,
-    )
+@flags_core.call_only_once
+def define_mnist_eager_flags():
+  """Defined flags and defaults for MNIST in eager mode."""
+  flags_core.define_base_eager()
+  flags_core.define_image()
+  flags.adopt_module_key_flags(flags_core)
```

Comment:
I've read through flags_core, and I still don't understand what this is doing. What does this do, and can we somehow make that activity more clear?